src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -143,6 +146,15 @@ private function createType(DocType $type, bool $nullable, string $docType = nul
             return new Type(Type::BUILTIN_TYPE_ARRAY, $nullable, null, true, $collectionKeyType, $collectionValueType);
         }
 
+        if ($type instanceof PseudoType) {
+            switch (\get_class($type->underlyingType())) {
```

Comment:
could(should) use a `match` statement and `instanceof` imho

Reply:
I didn't use `match` because I don't define the default. It just falls back to the existing logic if the underlying type is not a string nor integer. I agree with your `instanceof` suggestion though and I refactored my code a little bit.